### PR TITLE
Deprecate SbStringCompareNoCase

### PR DESCRIPTION
**Pull request by madhurajayaraman**
b/305768669

### CODE REVIEW

File: cobalt/loader/cors_preflight.cc [hunk 1]
```diff
@@ -117,9 +117,15 @@ bool IsInArray(const char* input_str, const char* array[], size_t size) {
     return false;
   }
   for (size_t i = 0; i < size; ++i) {
+    #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/loader/cors_preflight.cc:120]: same

File: cobalt/loader/cors_preflight.cc [hunk 2]
```diff
@@ -132,10 +138,18 @@ bool HasFieldValue(const std::vector<std::string>& field_values,
     if (field_values[i].empty()) {
       continue;
     }
+    #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/loader/cors_preflight.cc:141]: same

File: cobalt/loader/cors_preflight.cc [hunk 3]
```diff
@@ -220,10 +234,17 @@ bool CORSPreflight::IsSafeResponseHeader(
   }
 
   for (size_t i = 0; i < CORS_exposed_header_name_list.size(); i++) {
+    #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/loader/cors_preflight.cc:237]: same

File: cobalt/loader/cors_preflight.cc [hunk 4]
```diff
@@ -381,12 +402,22 @@ void CORSPreflight::OnURLFetchComplete(const net::URLFetcher* source) {
       if (HasFieldValue(headernames_vec, it.name())) {
         continue;
       }
+      #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/loader/cors_preflight.cc:405]: same

File: cobalt/loader/cors_preflight_cache.cc [hunk 5]
```diff
@@ -127,10 +127,17 @@ bool CORSPreflightCache::HaveEntry(
   if (entry_ptr->allow_all_headers_except_non_wildcard) {
     bool has_auth_header = false;
     for (const auto& header : unsafe_headernames) {
+      #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/loader/cors_preflight_cache.cc:130]: same

File: cobalt/loader/cors_preflight_cache.h [hunk 6]
```diff
@@ -54,7 +54,11 @@ class CORSPreflightCache : public base::RefCounted<CORSPreflightCache> {
   // Case-insensitive comparator.
   struct CaseInsensitiveCompare {
     bool operator()(const std::string& lhs, const std::string& rhs) const {
+      #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/loader/cors_preflight_cache.h:57]: same

File: cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontMgr_cobalt.cc [hunk 7]
```diff
@@ -307,9 +307,15 @@ void SkFontMgr_Cobalt::LoadLocaleDefault() {
   std::string script =
       icu::Locale::createCanonical(base::GetSystemLanguageScript().c_str())
           .getScript();
+  #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontMgr_cobalt.cc:310]: same

File: cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontMgr_cobalt.cc [hunk 8]
```diff
@@ -515,9 +521,15 @@ bool SkFontMgr_Cobalt::CheckIfFamilyMatchesLocaleScript(
   }
   std::string family_script =
       icu::Locale::createCanonical(family_tag.c_str()).getScript();
+  #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontMgr_cobalt.cc:524]: same

File: cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontStyleSet_cobalt.cc [hunk 9]
```diff
@@ -232,6 +232,7 @@ SkFontStyleSet_Cobalt::SkFontStyleSet_Cobalt(
     ++extension;
 
     // Only add font formats that match the format setting.
+    #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontStyleSet_cobalt.cc:235]: same

File: cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontStyleSet_cobalt.cc [hunk 10]
```diff
@@ -272,13 +284,23 @@ SkFontStyleSet_Cobalt::SkFontStyleSet_Cobalt(
     int* index = styles_index_map.find(font_name);
     if (index != nullptr) {
       // If style with name already exists in family, replace it.
+      #if SB_API_VERSION < 16
```
Comment (y4vor) [inline cobalt/renderer/rasterizer/skia/skia/src/ports/SkFontStyleSet_cobalt.cc:287]: same

File: starboard/client_porting/poem/strings_poem.h [hunk 11]
```diff
@@ -23,10 +23,12 @@
 
 #include "starboard/string.h"
 
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline starboard/client_porting/poem/strings_poem.h:26]: lets delete the whole section

**Comment by y4vor** [comment]
for the stub failures, please add the symbol to the list of posix symbols:
https://github.com/youtube/cobalt/blob/0ee9237ab29ce7defd8f9d96cae5522abb0b6920/starboard/tools/api_leak_detector/api_leak_detector.py#L89

And for win32 you need to define something like
#define stricmp _stricmp 

in starboard/shared/win32/posix_emu/include/stdlib.h

**Review by y4vor — APPROVED** [review]

**Review by y4vor — APPROVED** [review]

File: starboard/elf_loader/exported_symbols.cc [hunk 12]
```diff
@@ -303,13 +303,11 @@ ExportedSymbols::ExportedSymbols() {
   REGISTER_SYMBOL(SbStorageOpenRecord);
   REGISTER_SYMBOL(SbStorageReadRecord);
   REGISTER_SYMBOL(SbStorageWriteRecord);
+#if SB_API_VERSION < 16
   REGISTER_SYMBOL(SbStringCompareNoCase);
```
Comment (andrewsavage1) [inline starboard/elf_loader/exported_symbols.cc:307]: Are we not going to now register strcasecmp and strncasecmp?

Comment (y4vor) [reply starboard/elf_loader/exported_symbols.cc:307]: We don't need them registered as we were able to use the musl free standing implementation. Take a look at the changes in third_party/musl/BUILD.gn.

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**